### PR TITLE
feat: Update test scripts use JSON for deployed libs (SC-2306)

### PR DESCRIPTION
**Pull request by lucas-manuel**
# Description

Test suite runs faster with this change, since we don't have to deploy libraries on every run.

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,243 @@
+{
+  "language": "Solidity",
```
Comment (satyamakgec) [inline config.json:2]: Can we update the readme about how to generate this file as the developer setup.

File: test.sh [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
+export DAPP_TEST_NUMBER=12543537
 export DAPP_SOLC_VERSION=0.6.11
 export DAPP_SRC="contracts"
-export DAPP_BUILD_OPTIMIZE=1
+export DAPP_BUILD_OPTIMIZE=0
 export DAPP_BUILD_OPTIMIZE_RUNS=200
-export DAPP_LINK_TEST_LIBRARIES=1
+export DAPP_LINK_TEST_LIBRARIES=0
 export DAPP_REMAPPINGS=$(cat remappings.txt)
+export DAPP_LIBRARIES=" contracts/libraries/loan/v1/LoanLib.sol:LoanLib:0x51A189ccD2eB5e1168DdcA7e59F7c8f39AA52232 contracts/libraries/pool/v1/PoolLib.sol:PoolLib:0x2c1C30fb8cC313Ef3cfd2E2bBf2da88AdD902C30 contracts/libraries/util/v1/Util.sol:Util:0x95f9676A34aF2675B63948dDba8F8c798741A52a"
```
Comment (satyamakgec) [inline test.sh]: we can re-use the build.sh here again to avoid confusion.

File: build.sh [hunk 3]
```diff
@@ -1,11 +1,14 @@
 #!/usr/bin/env bash
 set -e
 
+export DAPP_TEST_TIMESTAMP=1622483493
+export DAPP_TEST_NUMBER=12543537
 export DAPP_SOLC_VERSION=0.6.11
 export DAPP_SRC="contracts"
-export DAPP_BUILD_OPTIMIZE=1
+export DAPP_BUILD_OPTIMIZE=0
```
Comment (deluca-mike) [inline build.sh:8]: If `optimize` is disabled, then I'm pretty sure `runs` does not matter, correct?

File: contracts/test/TestUtil.sol [hunk 4]
```diff
@@ -585,6 +585,22 @@ contract TestUtil is DSTest {
...
+
+        uint256 denominator = val0 == 0 ? val1 : val0;
+        bool check = ((diff * RAY) / denominator) < percentage * RAY / 10_000;
+
+        if (!check){
+            emit log_named_uint("Error: approx a == b not satisfied, accuracy digits ", percentage);
+            emit log_named_uint("  Expected", val0);
+            emit log_named_uint("    Actual", val1);
+            fail();
+        }
```
Comment (deluca-mike) [inline contracts/test/TestUtil.sol]:
```suggestion
        if (check) return;
        
        emit log_named_uint("Error: approx a == b not satisfied, accuracy digits ", percentage);
        emit log_named_uint("  Expected", val0);
        emit log_named_uint("    Actual", val1);
        fail();
```

File: test.sh [hunk 5]
```diff
@@ -1,31 +1,34 @@
...
 [[ $SKIP_MAINNET_CHECK || "$ETH_RPC_URL" && "$(seth chain)" == "ethlive" ]] || { echo "Please set a mainnet ETH_RPC_URL"; exit 1; }
 
-export DAPP_TEST_TIMESTAMP=1615792486
-export DAPP_TEST_NUMBER=12045000
+export DAPP_TEST_TIMESTAMP=1622483493
+export DAPP_TEST_NUMBER=12543537
 export DAPP_SOLC_VERSION=0.6.11
 export DAPP_SRC="contracts"
-export DAPP_BUILD_OPTIMIZE=1
+export DAPP_BUILD_OPTIMIZE=0
```
Comment (deluca-mike) [inline test.sh]: Ditto on `runs` being ignored if `optimize` disabled.

**Review by deluca-mike — CHANGES_REQUESTED** [review]

File: test.sh (hunk 2 above)
Comment (lucas-manuel) [reply test.sh]: I wanted to keep them separate so we can have a separate job in CI to run the build

**Review by satyamakgec — APPROVED** [review]

**Review by deluca-mike — APPROVED** [review]

**Pull request merged**